sdcm/cluster_k8s/__init__.py

```diff
@@ -989,39 +1057,54 @@ def scylla_machine_image_args_modifier(obj):
                 if "scylla-machine-image:" in container_data["image"]:
                     container_data["args"] = scylla_machine_image_args
 
-        self.apply_file(
-            self.NODE_PREPARE_FILE,
-            modifiers=[affinity_modifier
-                       for current_pool in node_pools
-                       for affinity_modifier in current_pool.affinity_modifiers
-                       ] + [scylla_machine_image_args_modifier],
-            envsubst=False)
-        self.install_static_local_volume_provisioner(node_pools=node_pools)
+        def node_config_remove_local_disk_setup(obj):
+            if obj["kind"] != "NodeConfig":
+                return
+            obj["spec"].pop("localDiskSetup", None)
+
+        def node_setup_for_dynamic_local_volume_provisioner_modifier(obj):
+            if obj["kind"] != "DaemonSet":
+                return
+            for container_data in obj["spec"]["template"]["spec"]["containers"]:
+                if container_data["name"] in ("pv-setup", "node-setup"):
+                    # NOTE: disable custom node- and pv- setups using dynamic volume provisioner
```

Comment:
won't it be better to keep separate files for each option ?

Reply:
For the second new option (`dynamic`) we download files from the github repo. It is not kept here.